crates/matrix-sdk/src/client/mod.rs

```diff
@@ -125,7 +128,7 @@ pub(crate) struct ClientInner {
     /// User session data.
     base_client: BaseClient,
     /// The Matrix versions the server supports (well-known ones only)
-    server_versions: Arc<[MatrixVersion]>,
+    server_versions: Mutex<Arc<[MatrixVersion]>>,
```

Comment:
I think this could be a prime candidate for a `RwLock`, no? We write once but read many times, and reads don't need to block other reads.

Reply:
That would make the code more complicated, bring weird edge cases around multiple threads observing the versions not being initialized yet at the same time, and be irrelevant for performance since after initialization, the duration we're holding the lock for is exceedingly small (all we there do is `Arc::clone`).

Reply:
Alright, let's leave it as is then.